### PR TITLE
Cleanup pods when job is complete

### LINKED ISSUE #51: Cleanup eraser pods after completed
- [x] #66
- [x] #67

### LINKED ISSUE #67: after pod is completed and status is received, delete pod

### LINKED ISSUE #90: Handle imagejob deletion gracefully
Right now if an imagejob is set to be deleted (`imageJob.DeletionTimestamp > 0`) we may miss updating image statuses.
Additionally it may leave behind running pods.

### PR DESCRIPTION
**Pull request by cpuguy83**
**What this PR does / why we need it**:

When a job is complete, clean up all pods associated with it as well.
This made it neccessary to add the deletecollection permission to the
cluster user (or else we need to delete one by one).

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Closes #90
Closes #51
Closes #67

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -93,7 +93,7 @@ test: manifests generate fmt vet envtest ## Run tests.
 # Run e2e tests
 .PHONY: e2e-test
 e2e-test:
-	IMAGE=${ERASER_IMG} MANAGER_IMAGE=${IMG} NODE_VERSION=kindest/node:v${KUBERNETES_VERSION} go test -tags=e2e -v ./test/e2e
+	IMAGE=${ERASER_IMG} MANAGER_IMAGE=${IMG} NODE_VERSION=kindest/node:v${KUBERNETES_VERSION} go test -count=1 -tags=e2e -v ./test/e2e
```
Comment (sozercan) [inline Makefile:96]: 👍

File: config/rbac/role.yaml [hunk 2]
```diff
@@ -1,106 +1,106 @@
...
+    resources:
+      - pods
+    verbs:
+      - create
+      - delete
+      - get
+      - list
+      - update
+      - watch
+      - deletecollection
```
Comment (sozercan) [inline config/rbac/role.yaml]: TIL

File: controllers/imagejob/imagejob_controller.go [hunk 3]
```diff
@@ -152,148 +153,188 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
-		for i := range nodes.Items {
-			nodeName := nodes.Items[i].Name
-			runtime := nodes.Items[i].Status.NodeInfo.ContainerRuntimeVersion
-			runtimeName := strings.Split(runtime, ":")[0]
-			mountPath := getMountPath(runtimeName)
-			if mountPath == "" {
-				log.Println("Incompatible runtime on node ", nodeName)
-				continue
-			}
+	// TODO: What to do if imageJob.DeletionTimestamp is set but jobs are not completed?
```
Comment (sozercan) [inline controllers/imagejob/imagejob_controller.go]: can you open an issue for tracking this if it's not going to be addressed in this pr (applies to other TODOs too)

**Review by sozercan — APPROVED** [review]
comment above, otherwise LGTM. Thank you!

**Comment by sozercan** [comment]
looks like manifest needs to be regenerated

File: controllers/imagejob/imagejob_controller.go [hunk 4]
```diff
@@ -152,148 +153,188 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
-				NodeName:           nodeName,
-				Volumes:            []v1.Volume{{Name: runtimeName + "-sock-volume", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: mountPath}}}},
-			}
+func (r *Reconciler) handleCompletedJob(ctx context.Context, imageJob *eraserv1alpha1.ImageJob) error {
+	gracePeriod := int64(0)
+	return r.DeleteAllOf(ctx, &v1.Pod{}, &client.DeleteAllOfOptions{
+		DeleteOptions: client.DeleteOptions{
+			GracePeriodSeconds: &gracePeriod,
+		},
+		// TODO: Is this list selection refined enough? Could this stomp on other imagejobs?
```
Comment (sozercan) [inline controllers/imagejob/imagejob_controller.go]: would it be better if we deleted the jobs here and that will delete the pods?

File: controllers/imagejob/imagejob_controller.go (hunk 3 above)
Comment (cpuguy83) [reply controllers/imagejob/imagejob_controller.go]: https://github.com/Azure/eraser/issues/90

File: controllers/imagejob/imagejob_controller.go (hunk 4 above)
Comment (cpuguy83) [reply controllers/imagejob/imagejob_controller.go]: Yep, I like it. Done.

**Pull request merged**